javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/utils/JSSCollectionStrategy.java

```diff
@@ -0,0 +1,80 @@
+package com.github.javaparser.symbolsolver.utils;
+
+import com.github.javaparser.ParserConfiguration;
+import com.github.javaparser.symbolsolver.JavaSymbolSolver;
+import com.github.javaparser.symbolsolver.resolution.typesolvers.CombinedTypeSolver;
+import com.github.javaparser.symbolsolver.resolution.typesolvers.JarTypeSolver;
+import com.github.javaparser.symbolsolver.resolution.typesolvers.JavaParserTypeSolver;
+import com.github.javaparser.symbolsolver.resolution.typesolvers.ReflectionTypeSolver;
+import com.github.javaparser.utils.CollectionStrategy;
+import com.github.javaparser.utils.ProjectRoot;
+
+import java.io.IOException;
+import java.nio.file.*;
+import java.nio.file.attribute.BasicFileAttributes;
+
+import static java.nio.file.FileVisitResult.CONTINUE;
+import static java.nio.file.FileVisitResult.SKIP_SUBTREE;
+
+/**
+ * Strategy which collects all SourceRoots and initialises the TypeSolver and returns the SourceRoots configured
+ * with the TypeSolver in a ProjectRoot object.
+ */
+public class JSSCollectionStrategy implements CollectionStrategy {
```

Comment:
I'd say write these class names without abbreviating. We're not abbreviating names much in the rest of the code.